orderer/consensus/etcdraft/node.go

```diff
@@ -57,8 +58,8 @@ func (n *node) start(fresh, join bool) {
 	var campaign bool
 	if fresh {
 		if join {
-			raftPeers = nil
 			n.logger.Info("Starting raft node to join an existing channel")
+			n.Node = raft.RestartNode(n.config)
```

Comment:
As per my understanding, fresh + join condition is equal to `adding a new node to existing channel`, it should be StartNode, why it is updated to RestartNode?

Reply:
The issue is how to add the node to its existing set of peers and config. The docs on the method explains this:
https://github.com/atoulme/fabric/blob/939220f920153ef4f8ac81833f07ad9c85760ebb/vendor/go.etcd.io/etcd/raft/v3/node.go#L238

This commit shows how state is recreated: https://github.com/hyperledger/fabric/pull/2997/commits/6ee57eb420eaaf46704c499ee01f42af648717f0